keps/sig-node/5018-dra-adminaccess/README.md

```diff
@@ -0,0 +1,691 @@
+// handle the request.
+type DeviceRequest struct {
+	...
+
+    // AdminAccess indicates that this is a claim for administrative access
+    // to the device(s). Claims with AdminAccess are expected to be used for
+    // monitoring or other management services for a device.  They ignore
+    // all ordinary claims to the device with respect to access modes and
+    // any resource allocations.
+    //
+    // This is an alpha field and requires enabling the DRAAdminAccess
+    // feature gate. Admin access is disabled if this field is unset or
+    // set to false, otherwise it is enabled.
+    //
+    // +optional
+    // +featureGate=DRAAdminAccess
+    AdminAccess *bool
+}
+```
+
+Admin access to devices is a privileged operation because it grants users access
+to devices that are in use by other users. Drivers might also remove other
+restrictions when preparing the device.
+
+In Kubernetes 1.31 (before this KEP was introduced), an example validating
+admission policy
+[was provided](https://github.com/kubernetes/kubernetes/blob/4aeaf1e99e82da8334c0d6dddd848a194cd44b4f/test/e2e/dra/test-driver/deploy/example/admin-access-policy.yaml#L1-L11)
+which restricts access to this option. It is the responsibility of cluster
+admins to ensure that such a policy is installed if the cluster shouldn't allow
+unrestricted access.
+
+Starting in Kubernetes 1.33 (when this KEP was introduced), a validation has
+been added to the REST storage layer to only authorize `ResourceClaim` or
+`ResourceClaimTemplate` with `adminAccess: true` requests if their namespace has
+the `kubernetes.io/dra-admin-access` label to only allow it for users with
+additional privileges. More time is needed to figure out how that should work,
+therefore the field is placed behind the `DRAAdminAccess` feature gate.
+
+The `DRAAdminAccess` feature gate controls whether users can set the
+`adminAccess` field to true when requesting devices. That is checked in the
+apiserver. In addition, the scheduler will not allocate claims with admin access
+when the feature is turned off. This covers the scenario that the field might
+have been set in Kubernetes v1.31 where the field was available uncondiionally
+or while the feature was enabled.
+
+### REST Storage Changes
+
+In both pkg/registry/resource/resourceclaim/strategy.go and
+pkg/registry/resource/resourceclaimtemplate/strategy.go, update `Validate` and
+`ValidateUpdate` to call a new `authorizedForAdmin` function to check if the
+request is authorized to set the `adminAccess` field to devices based on the DRA
+admin namespace label.
+
+### Kube-controller-manager Changes
+
+In pkg/controller/resourceclaim/controller.go, process `ResourceClaim` in
+`syncClaim` function to check for the `adminAccess` field and the feature gate
+enablement to ensure the field can be set.
+
+// TODO: what part of claim.Status.Allocation should be updated? e.g.
+AdminAccess is part of `DeviceRequestAllocationResult` but need to set it for
+each device?
```

Comment:
@pohly any thoughts on this?

Reply:
The [current API](https://github.com/kubernetes/kubernetes/blob/c5c19a64877310d4350ada3bf2428654c8117888/staging/src/k8s.io/api/resource/v1beta1/types.go#L832-L842) is correct: the fact that admin access was used during allocation must be recorded for each device where that mode was applied.

Reply:
I will remove any mentions of syncClaim then. Is the device plugin expected to update AdminAccess for each applicable device in `DeviceRequestAllocationResult`?

Reply:
The scheduler sets `DeviceRequestAllocationResult.AdminAccess` when it allocates a claim. A DRA driver reads that field.

"device plugin" is a different kind of kubelet plugin, not something that is used by DRA.